akka-http/src/main/scala/akka/http/scaladsl/server/directives/HeaderDirectives.scala

```diff
@@ -133,7 +133,7 @@ trait HeaderDirectives {
    * @group header
    */
   def optionalHeaderValueByName(headerName: String): Directive1[Option[String]] = {
-    val lowerCaseName = headerName.toLowerCase
+    val lowerCaseName = headerName.toRootLowerCase
```

Comment:
This makes sense to me, though I wonder if there aren't more places where we should do this

Reply:
Definitely, but that one was related to the previous PR, so makes sense to do that here.